src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -452,8 +452,12 @@ export class SearchView extends ViewletPanel {
 					return;
 				}
 
-				const root = element instanceof SearchResult ? null : element;
-				this.tree.setChildren(root, this.createIterator(element, collapseResults));
+				if (element instanceof SearchResult) {
+					this.tree.setChildren(null, this.createIterator(element, collapseResults));
+				} else {
+					this.tree.setChildren(element, this.createIterator(element, collapseResults));
+					this.tree.rerender(element);
```

Comment:
Why does this fix it, shouldn't it rerender already when setChildren is called?

Reply:
@roblourens from what I understand from [here](https://github.com/Microsoft/vscode/blob/6d444df6669feb905909e636741bd7a770ed570a/src/vs/base/browser/ui/tree/indexTreeModel.ts#L89-L188) to refresh the parent node you have to spliced it from the list as in rerender `this.list.splice(listIndex, 1, [node]);` but the splice function from indexTreeModel just does it for the [children](https://github.com/Microsoft/vscode/blob/6d444df6669feb905909e636741bd7a770ed570a/src/vs/base/browser/ui/tree/indexTreeModel.ts#L161).

Reply:
Oh I see , thanks